Make FIFO files more secure

avilcheslopez opened the pull request:
- FIFO files are now created in geopm's runtime directory to avoid
security vulnerabilities related to using the tmp directory
- Documentation and tests updated accordingly

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

- Relates to #2136 
- Fixes #2210

Move FIFO files to geopm's runtime directory to make them more secure.

avilcheslopez committed:
Make FIFO files more secure

- FIFO files are now created in geopm's runtime directory to avoid
security vulnerabilities related to using the tmp directory
- Documentation and tests updated accordingly

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

cmcantalupo commented on service/docs/source/security.rst:
```diff
@@ -472,11 +472,19 @@ When making directories, if the path already exists checks are
+only for the owner). Some directories, for example
+``/var/run/geopm-service``, also require execution permissions (i.e.
+0o711). For more details on how directories are created and default
+permissions, please see the `systme_files.py <http://geopm.github.io/geopmdpy.7.html#module-geopmdpy.system_files>`__
```
```suggestion
permissions, please see the `system_files.py <http://geopm.github.io/geopmdpy.7.html#module-geopmdpy.system_files>`__
```

avilcheslopez replied:
Fixed.

cmcantalupo commented on service/geopmdpy_test/TestActiveSessions.py:
```diff
@@ -168,7 +169,9 @@ def check_json_file(self, name, contents, is_valid):
              mock.patch('sys.stderr.write', return_value=None) as mock_err:
             act_sess = ActiveSessions(sess_path)
-            mock_smd.assert_called_once_with(sess_path)
+            mock_smd.assert_called_once_with(
+                sess_path,
+                perm_mode=GEOPM_SERVICE_VAR_PATH_PERM)
```
```suggestion
            mock_smd.assert_called_once_with(sess_path,
                perm_mode=GEOPM_SERVICE_VAR_PATH_PERM)
```

avilcheslopez committed:
Avoid exposing FIFO implementation details to BatchStatus clients/users.

- Moved default FIFO path const to the implementation class
- Removed FIFO path details from BatchStatus interface
  (make_unique functions)

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

cmcantalupo commented on service/src/BatchStatus.hpp:
```diff
@@ -92,6 +91,11 @@ namespace geopm
+            // This is the single place where the server prefix is located,
+            // which is also accessed by BatchStatusTest.
+            static constexpr const char* M_DEFAULT_FIFO_PREFIX =
+                "/var/run/geopm-service/geopm-service-batch-status-";
```
```suggestion
                "/var/run/geopm-service/batch-status-";
```

cmcantalupo commented on service/test/BatchStatusTest.cpp:
```diff
@@ -59,7 +65,7 @@ class BatchStatusTest : public ::testing::Test
 void BatchStatusTest::SetUp(void)
 {
-    m_server_prefix = BatchStatus::M_FIFO_PREFIX;
+    m_server_prefix = "/tmp/geopm-service-batch-status-";
```
```suggestion
    m_server_prefix = "/tmp/geopm-test-service-batch-status-";
```

cmcantalupo reviewed (commented):
Just a couple of name change suggestions on file paths.  Otherwise looks great!  Thanks for hiding the fifo path from the public interface.

avilcheslopez committed:
Minor clean-up, add unit test for secure dirs interface

- Renamed default FIFO file prefix
- Minor fix in service README documentation
- Added a unit test for secure_make_dirs

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

cmcantalupo commented on service/geopmdpy_test/TestSecureFiles.py:
```diff
@@ -106,6 +106,19 @@ def test_default_creation(self):
         secure_make_dirs(sess_path)
         self.check_dir_perms(sess_path)
 
+    def test_creation_with_perm(self):
```
You'll need to add this test fixture name to the Makefile.mk in this directory to have it run with `make check`

avilcheslopez committed:
Enable new unit test in TestSecureFiles.py

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

cmcantalupo reviewed (approved):
👍
